File: Bidapp Demo App - ObjectiveC/bidappPlayground/FullscreenShowDelegate.m
//
//  FullscreenShowDelegate.m
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

#import "FullscreenShowDelegate.h"
#import <bidapp/BIDAdInfo.h>
#import "tools.h"

@interface FullscreenShowDelegate()
{
	UIViewController* viewController;
}

@property(nonatomic) NSString* sessionId;
@property(nonatomic) NSString* waterfallId;

@end

@implementation FullscreenShowDelegate

-(id)initWithViewController:(UIViewController*)vc
{
	if (self = [super init])
	{
		viewController = vc;
	}
	
	return self;
}

#pragma mark - BIDInterstitialDelegate and BIDRewardedDelegate

- (nonnull UIViewController*)viewControllerForDisplayAd
{
	return viewController;
}

- (void)didDisplayAd:(BIDAdInfo*)adInfo
{
	_sessionId = adInfo.showSessionId;
	_waterfallId = adInfo.waterfallId;

	NSLog(@"[%@][%@] didDisplayAd: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
}

- (void)didClickAd:(BIDAdInfo*)adInfo
{
	NSLog(@"[%@][%@] didClickAd: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
}

- (void)didHideAd:(BIDAdInfo*)adInfo
{
	NSLog(@"[%@][%@] didHideAd: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
}

- (void)didFailToDisplayAd:(BIDAdInfo*)adInfo error:(NSError *)error
{
	_sessionId = adInfo.showSessionId;
	_waterfallId = adInfo.waterfallId;
    
	NSLog(@"[%@][%@] didFailToDisplayAd: %d, ERROR: %@", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId, error.localizedDescription ? error.localizedDescription : error);
}

-(void)allNetworksDidFailToDisplayAd
{
	NSLog(@"[%@][%@] allNetworksDidFailToDisplayAd", sessionIdShort(_sessionId), _waterfallId);
}

#pragma mark - BIDRewardedDelegate

- (void)didRewardUser
{
	NSLog(@"[%@][%@] didRewardUser",sessionIdShort(_sessionId), _waterfallId);
}

@end

File: Bidapp Demo App - Swift/bidappPlayground/FullscreenShowDelegate.swift
//
//  FullscreenShowDelegate.swift
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

import UIKit
import bidapp

class FullscreenShowDelegate : NSObject, BIDRewardedDelegate, BIDInterstitialDelegate {
    
    let viewController:UIViewController
    var sessionId:String = ""
    var waterfallId:String = ""
    
    init(viewController:UIViewController) {
        self.viewController = viewController
    }
    
    func viewControllerForDisplayAd() -> UIViewController {
        return viewController
    }
    
    func didDisplayAd(_ adInfo: BIDAdInfo) {
        sessionId = adInfo.showSessionId ?? ""
        sessionId = String(sessionId.prefix(3))
        waterfallId = adInfo.waterfallId ?? ""
        
        print("Bidapp fullscreen [\(sessionId)][\(waterfallId)] didDisplayAd: \(adInfo.networkId)")
    }
    
    func didClickAd(_ adInfo: BIDAdInfo) {
        print("Bidapp fullscreen [\(sessionId)][\(waterfallId)] didClickAd: \(adInfo.networkId)")
    }
    
    func didHideAd(_ adInfo: BIDAdInfo) {
        print("Bidapp fullscreen [\(sessionId)][\(waterfallId)] didHideAd: \(adInfo.networkId)")
    }
    
    func didFail(toDisplayAd adInfo: BIDAdInfo, error: Error) {
        sessionId = adInfo.showSessionId ?? ""
        sessionId = String(sessionId.prefix(3))
        waterfallId = adInfo.waterfallId ?? ""
        
        print("Bidapp fullscreen [\(sessionId)][\(waterfallId)] didFailToDisplayAd: \(adInfo.networkId), ERROR: \(error.localizedDescription)")
    }
    
    func allNetworksDidFailToDisplayAd() {
        print("Bidapp fullscreen [\(sessionId)][\(waterfallId)] allNetworksDidFailToDisplayAd")
    }
    
    func didRewardUser() {
        print("Bidapp fullscreen didRewardUser")
    }
}
